packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +134,42 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      return getODataFilterExpressionWhenBeingFilterLambdaExpression(
+        filter,
+        parentFieldNames,
+        targetEntityConstructor
+      );
+    }
+  }
+
+  function getODataFilterExpressionWhenBeingFilterLambdaExpression<
+    FilterEntityT extends EntityBase
+  >(
+    filter: FilterLambdaExpression<FieldType>,
+    parentFieldNames: string[] = [],
+    targetEntityConstructor: Constructable<any>
+  ) {
+    const alias = 'a';
+    if (typeof filter.innerFilter.field !== 'string') {
+      throw new Error(
+        `The type of the field: ${
+          filter.innerFilter.field
+        } is not string, but ${typeof filter.innerFilter.field}.`
+      );
+    }
+    const field = retrieveField(
+      filter.innerFilter.field,
+      targetEntityConstructor,
+      filter.innerFilter.edmType
+    );
+    const filterExp = [
+      [...parentFieldNames, filter.innerFilter.field].join('/'),
+      filter.innerFilter.operator,
+      uriConverter.convertToUriFormat(filter.innerFilter.value, field.edmType)
+    ].join(' ');
+    return `${filter.rootFiledName}/${filter.lambdaOperator}(${alias}:${alias}/${filterExp})`;
```

Comment:
General: What happens if I go down two or more layers:
TestEntity.TO_MULTILINK_1.TO_MULTILINK_2.filter(). Then the `this` in the `filter()` points to the TO_MULTILINK_2  which is then the rootFiledName. Is this correct or should on recursivly go up to the root entity to build a string containing all links?

Reply:
2 and more links are not possible just like the example here:
`$filter=Friends/Friends/any(f:f/UserName eq 'scottketchum')`, which give you a 400 with message `"The parent value for a property access of a property 'Friends' is not a single value. Property access can only be applied to a single value."`
I'll then throw an error when 2+  links are used.